### PR TITLE
Fixed #27583 -- MultiValueDict.getlist crash when values for key is None.

### PR DESCRIPTION
**Pull request by felixxm**
[Ticket](https://code.djangoproject.com/ticket/27583).

### CODE REVIEW

**Comment by timgraham** [comment]
The behavior in Django 1.10 and earlier is to return `None` rather than an empty list -- might be better to preserve that (crashing regression was introduced in 727d7ce6cba21363470aaefb2dc5353017531be3).

**Comment by felixxm** [comment]
I think you're right. I updated PR.

File: tests/utils_tests/test_datastructures.py [hunk 1]
```diff
@@ -119,6 +119,11 @@ def test_getlist_default(self):
         values = x.getlist('b', default=MISSING)
         self.assertIs(values, MISSING)
 
+    def test_getlist_none_values(self):
+        x = MultiValueDict({'a': None})
+        values = x.getlist('a')
+        self.assertEqual(values, None)
```
Comment (timgraham) [inline tests/utils_tests/test_datastructures.py]: `self.assertIsNone(x.getlist('a'))` seems okay.

**Comment by felixxm** [comment]
Thanks for review. Sorry for updating in a hurry.

**Pull request merged**